scripts/validate-tokens/src/main.rs

```diff
@@ -0,0 +1,193 @@
+use alloy::primitives::Address;
+use alloy::providers::{ProviderBuilder, RootProvider};
+use alloy::sol;
+use alloy::transports::http::reqwest::Url;
+use alloy::transports::http::{Client, Http};
+use base64::{engine::general_purpose, Engine as _};
+use serde::Deserialize;
+use std::env;
+use std::fs;
+use std::str::FromStr;
+
+sol! {
+    #[sol(rpc)]
+    interface IERC20 {
+        function name() view returns (string);
+        function symbol() view returns (string);
+        function decimals() view returns (uint8);
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Token {
+    address: String,
+    symbol: String,
+    name: String,
+    decimals: u8,
+}
+
+#[derive(Deserialize)]
+struct TokenInfoResponse {
+    data: TokenInfo,
+}
+
+#[derive(Deserialize)]
+struct TokenInfo {
+    name: String,
+    symbol: String,
+    decimals: u8,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
I would recommend returning an error from main if any token is incorrect. Then you can run this as a check in CI and get an non-zero exit code to make the testrun fail.